components/zcash_address/src/kind/unified.rs

```diff
@@ -286,15 +286,24 @@ pub(crate) mod private {
         /// items in their given order.
         fn try_from_items_internal(items: Vec<Self::Item>) -> Result<Self, ParseError> {
             let mut typecodes = HashSet::with_capacity(items.len());
+            let mut prev_typecode = None;
             for item in &items {
                 let t = item.typecode();
                 if typecodes.contains(&t) {
                     return Err(ParseError::DuplicateTypecode(t));
+                } else if prev_typecode
+                    .iter()
+                    .any(|tc| <u32>::from(*tc) > <u32>::from(t))
+                {
+                    return Err(ParseError::InvalidEncoding(
+                        "Receivers out of order.".to_owned(),
```

Comment:
```suggestion
                        "Items out of order.".to_owned(),
```
(This is used for UVKs as well.)